evaluator.go

```diff
@@ -73,12 +78,24 @@ func (s *State) Clusters() map[string]Cluster {
 	return s.clusters
 }
 
+// enable clusterQueryResultCaching
```

Comment:
was concerned about increase in memory requirements for caching results. Caching will be an opt in feature at expense of little memory increase